shared/video-sdk/develop/ensure-channel-quality/project-implementation/unity.mdx

```diff
@@ -0,0 +1,259 @@
+    * `OnLastmileProbeResult`: Receives detailed probe test results.
+    * `OnNetworkQuality`: Receives statistics on network quality.
+    * `OnRtcStats`: Receives the <Vg k="ENGINE" /> stats.
+    * `OnRemoteVideoStateChanged`: Receives notification regarding any change in the state of the remote video.
+    * `OnRemoteVideoStats`: Receives stats about the remote videos.
+
+    In `Assets/Agora-RTC-Plugin/Agora-Unity-RTC-SDK/Code/JoinChannelVideo.cs`, add the following callback methods to the `UserEventHandler` class:
+
+    ```csharp
+    public override void OnConnectionStateChanged(RtcConnection connection, CONNECTION_STATE_TYPE state, CONNECTION_CHANGED_REASON_TYPE reason) {
+        Debug.Log("Connection state changed"
+                + "\n New state: " + state
+                + "\n Reason: " + reason);
+    }
+
+    public override void OnLastmileQuality(int quality) {
+        _videoSample.updateNetworkStatus(quality);
+    }
+
+    public override void OnLastmileProbeResult(LastmileProbeResult result) {   
+        _videoSample.RtcEngine.StopLastmileProbeTest();
+        Debug.Log("Probe test finished");
+        // The result object contains the detailed test results that help you
+        // manage call quality, for example, the downlink jitter.
+        Debug.Log("Downlink jitter: " + result.downlinkReport.jitter);
+    }
+
+    public override void OnNetworkQuality(RtcConnection connection, uint remoteUid, int txQuality, int rxQuality) {
+        // Use downlink network quality to update the network status
+        _videoSample.updateNetworkStatus(rxQuality);
+    }
+
+    public override void OnRtcStats(RtcConnection connection, RtcStats rtcStats) {
+        string msg = ""; 
+        msg = rtcStats.userCount + " user(s)";
+        msg = "Packet loss rate: " + rtcStats.rxPacketLossRate;
+        Debug.Log(msg);
+    }
+
+    public override void OnRemoteVideoStateChanged(RtcConnection connection, uint remoteUid, REMOTE_VIDEO_STATE state, REMOTE_VIDEO_STATE_REASON reason, int elapsed) {
+        string msg = "Remote video state changed: \n Uid =" + remoteUid
+                + " \n NewState =" + state
+                + " \n reason =" + reason
+                + " \n elapsed =" + elapsed;
+
+        Debug.Log(msg);
+    }
+    public override void OnRemoteVideoStats(RtcConnection connection, RemoteVideoStats stats) {
+        string msg = "Remote Video Stats: "
+                + "\n User id =" + stats.uid
+                + "\n Received bitrate =" + stats.receivedBitrate
+                + "\n Total frozen time =" + stats.totalFrozenTime;
+            Debug.Log(msg);
+    }
+    ```
+
+    Each event reports the statistics of the audio video streams from each remote user and host. 
+
+
+1. **Switch stream quality when the user taps the remote video**
+
+    To take advantage of dual-stream mode and switch remote video quality to high or low, add the following to the `JoinChannelVideo` class:
```

Comment:
Switch video mode and obtain the uid are two different things, can you update as steps please.